### PR TITLE
Fix CICD workflows - pre-build-checks, gh-event-notification, nightly-build

### PR DESCRIPTION
**Pull request by imda-kwokwk**
## Description

1. pre-build-checks - fix missing pip-audit=report.md, cleanup displayed messages
2. gh-event-notification -  
3. nightly-build - remove smoke test, rename to sit-build

## Motivation and Context

bug fix

## Type of Change

- fix: A bug fix

## How to Test

Run workflow manually in ci-workflow-dev branch.

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x]  I have tested the changes locally and verified that they work as expected.
- [x]  I have added or updated the necessary documentation (README, API docs, etc.).
- [ ]  I have added appropriate unit tests or functional tests for the changes made.
- [ ]  I have followed the project's coding conventions and style guidelines.
- [ ]  I have rebased my branch onto the latest commit of the main branch.
- [ ]  I have squashed or reorganized my commits into logical units.
- [ ]  I have added any necessary dependencies or packages to the project's build configuration.
- [x]  I have performed a self-review of my own code.
- [x]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>

### CODE REVIEW

**Review by imda-jacksonboey — DISMISSED** [review]
ok

File: .github/workflows/license-file-gen.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+name: License File Generation
```
Comment (imda-benedictlee) [inline .github/workflows/license-file-gen.yaml:1]: Need to change to "Notices File Generation"

File: .github/workflows/license-file-gen.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        default: 'aiverify-foundation/moonshot'
+        type: string
+      branch:
+        description: 'Branch or tag'
+        required: true
+        default: 'main'
+        type: string
+
+jobs:
+  license-file-gen:
```
Comment (imda-benedictlee) [inline .github/workflows/license-file-gen.yaml:31]: Need to change to "Notices File Gen"

File: .github/workflows/license-file-gen.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+          repository: ${{ github.event.inputs.repo }}
+          ref: ${{ github.event.inputs.branch }}
+          submodules: recursive
+
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Generate license file
```
Comment (imda-benedictlee) [inline .github/workflows/license-file-gen.yaml:47]: Need to change to "Generate Notices File"

File: .github/workflows/license-file-gen.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+          repo_name=$(echo "${{ github.event.inputs.repo }}" | awk -F'/' '{print $2}')
+          bash <(curl -s -L https://detect.synopsys.com/detect9.sh) \
+            --blackduck.url=https://blackduck.aipo-imda.net/ \
+            --blackduck.api.token=${{ secrets.BLACKDUCK_TOKEN }} \
+            --detect.detector.search.depth=10 \
+            --detect.project.name=aiverify-${repo_name} \
+            --detect.project.version.name=${{ github.event.inputs.branch }} \
+            --detect.source.path=. \
+            --detect.code.location.name="aiverify-${repo_name}_${{ github.event.inputs.branch }}" \
+            --detect.excluded.directories=/tests \
```
Comment (imda-benedictlee) [inline .github/workflows/license-file-gen.yaml:61]: Need to change to "--detect.excluded.directories=venv,tests"

**Review by imda-benedictlee — CHANGES_REQUESTED** [review]
Need to change License to Notices as License is already in the repo called Apache 2.0. We are generating Notices for the dependencies that we used.

File: .github/workflows/license-file-gen.yaml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            --blackduck.trust.cert=true \
+            --detect.excluded.detector.types=POETRY \
+            --detect.blackduck.scan.mode=RAPID \
+            --detect.notices.report=true \
+            --detect.notices.report.path=.
+          set -e
+          ls -al
+          cat ./*_Black_Duck_Notices_Report.txt
```
Comment (imda-benedictlee) [inline .github/workflows/license-file-gen.yaml:69]: Can I check how would the Notices file be committed to Main Branch?

**Comment by imda-benedictlee** [comment]
Hi @imda-kwokwk, I added some comments for the license-file-gen.yaml file.

**Comment by imda-kwokwk** [comment]
Changed to "Notices FIle Generation"

File: .github/workflows/license-file-gen.yaml (hunk 5 above)
Comment (imda-kwokwk) [reply .github/workflows/license-file-gen.yaml]: #
#      - name: Commit license file
#        run: |
#          set +e
#          mv aiverify_moonshot_main_Black_Duck_Notices_Report.txt LICENSE-2.md
#          git add LICENSE-2.md
#          git config user.name "imda-btg"
#          git config user.email "idma-btg@imda.gov.sg"
#          git commit -m "update license file"
#          git push
#          set -e
#

Previously, I have tessted the above commented off code to work.  However I have commented it off as we only generate the notices file on release.

File: .github/workflows/license-file-gen.yaml (hunk 4 above)
Comment (imda-kwokwk) [reply .github/workflows/license-file-gen.yaml]: Hi, not sure why the need to exclude venv, I thot Blackduck should be scanning the actual dependencies installed?

Comment (imda-benedictlee) [reply .github/workflows/license-file-gen.yaml]: ah I see....if thats the case, just need to change to tests. There was a typo on the exclusion on the regex for matching the folders name.

File: .github/workflows/license-file-gen.yaml (hunk 5 above)
Comment (imda-benedictlee) [reply .github/workflows/license-file-gen.yaml]: noted.

File: .github/workflows/notices-file-gen.yaml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        default: 'aiverify-foundation/moonshot'
+        type: string
+      branch:
+        description: 'Branch or tag'
+        required: true
+        default: 'main'
+        type: string
+
+jobs:
+  license-file-gen:
```
Comment (imda-benedictlee) [inline .github/workflows/notices-file-gen.yaml:31]: Change to "notices-file-gen":

File: .github/workflows/notices-file-gen.yaml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+          repository: ${{ github.event.inputs.repo }}
+          ref: ${{ github.event.inputs.branch }}
+          submodules: recursive
+
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Generate license file
```
Comment (imda-benedictlee) [inline .github/workflows/notices-file-gen.yaml:47]: Change to "Generate Notices File".

File: .github/workflows/notices-file-gen.yaml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            --detect.excluded.detector.types=POETRY \
+            --detect.blackduck.scan.mode=RAPID \
+            --detect.notices.report=true \
+            --detect.notices.report.path=.
+          set -e
+          ls -al
+          cat ./*_Black_Duck_Notices_Report.txt
+
+#
+#      - name: Commit license file
```
Comment (imda-benedictlee) [inline .github/workflows/notices-file-gen.yaml:72]: Change to "Commit Notices File"

File: .github/workflows/notices-file-gen.yaml [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+#        run: |
+#          set +e
+#          mv aiverify_moonshot_main_Black_Duck_Notices_Report.txt LICENSE-2.md
```
Comment (imda-benedictlee) [inline .github/workflows/notices-file-gen.yaml:75]: Change to "NOTICES.md"

File: .github/workflows/notices-file-gen.yaml [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+#          git add LICENSE-2.md
```
Comment (imda-benedictlee) [inline .github/workflows/notices-file-gen.yaml:76]: Change to "NOTICES.md"

File: .github/workflows/notices-file-gen.yaml [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+#          git config user.name "imda-btg"
+#          git config user.email "idma-btg@imda.gov.sg"
+#          git commit -m "update license file"
```
Comment (imda-benedictlee) [inline .github/workflows/notices-file-gen.yaml:79]: Change. to "git commit -m "Update Notices File"

**Review by imda-benedictlee — CHANGES_REQUESTED** [review]

File: .github/workflows/notices-file-gen.yaml (hunk 11 above)
Comment (imda-kwokwk) [reply .github/workflows/notices-file-gen.yaml:79]: Done

File: .github/workflows/notices-file-gen.yaml (hunk 10 above)
Comment (imda-kwokwk) [reply .github/workflows/notices-file-gen.yaml:76]: Done

File: .github/workflows/notices-file-gen.yaml (hunk 9 above)
Comment (imda-kwokwk) [reply .github/workflows/notices-file-gen.yaml:75]: Done

File: .github/workflows/notices-file-gen.yaml (hunk 8 above)
Comment (imda-kwokwk) [reply .github/workflows/notices-file-gen.yaml:72]: Done

File: .github/workflows/notices-file-gen.yaml (hunk 7 above)
Comment (imda-kwokwk) [reply .github/workflows/notices-file-gen.yaml:47]: Done

File: .github/workflows/notices-file-gen.yaml (hunk 6 above)
Comment (imda-kwokwk) [reply .github/workflows/notices-file-gen.yaml:31]: Done

**Review by imda-benedictlee — APPROVED** [review]
Approved

**Review by imda-jacksonboey — APPROVED** [review]
ok

**Pull request merged**